### DISCUSSION TITLE
WIFI to WIFI not working

### DISCUSSION THREAD

**Discussion by bhboyle** [RaspAP/raspap-webgui#1022] [Q&A] [answered]
Good afternoon all.

First let me say thank you for developing this software. It is exactly what I want for my RV. RV travel routers are STUPID expensive.  My use case is when I go RVing I would like to have one WIFI interface connect to the distant "local" wifi and rebroadcast it on my own local wifi. Perhaps even use a celular model in place of the external wifi when wifi is not an option.

That said, I am unable to get data to flow between the two wifi interfaces. I am working with a fresh 32it headless setup. doing the updates and setting locale then installing with the quick setup. Everything comes up but the hotspot is not ale to route traffic outside.  WAN0 shows up as no internet on the networking page.  I am not even able to ping anyhere if I ssh into the Raspbery pi.

In terns of hardware, I am using a Raspberry pi 4 2gig with an Panda Networks PAU09 external interface.

I have tried the same thing over and over hoping something different would happen (insanity anyone). 

Also it is difficult to search for wifi to wifi... so I have not found an existing solution.

Any help would be very welcome.

Thank you!

**Comment by aluminum-ice** [comment]
I have the same issue. I did a clean install on a Pi3B and verified that RaspAp worked correctly when I had eth0 plugged in and wlan0 as the AP. Then I unplugged eth0 and confirmed wlan1 was connected as a client to my router. I can ssh into the Pi via the wlan1 IP (assigned by the router 192.168.1.75) or via the AP IP (static 10.3.141.1) but no traffic is routed between wlan0 and wlan1. When I try ping -c3 www.cnn.con I get "Destination Host Unreachable". 

However, I can ping other machines on my internal network (Eg ping -c3 192.168.1.131 returns all the packets). That example pings my PiHole DNS which is on my network.

**Reply by aluminum-ice** [reply]
I figured out what is happening. The routing table has a default route that sends packets destined to the internet to the wrong gateway. To remove that default path:

sudo route del default gw 10.3.141.1

Now I need to determine if the default route reappears with every reboot. If it does, I'm not sure how to permanently resolve this issue... would love some help on that.

**Reply by aluminum-ice** [reply]
I semi-permanently fixed this by editing /etc/dhcpcd.conf to add (metric 500) to reduce the precedence of the incorrect default route. So, the route is still there but with lower precedence which enables the routing to work between reboots. 

interface wlan0
...
...
metric 500 # Reduce the precedence of this route

**Comment by billz** [comment]
Metric values can be set for any available interface in **DHCP Server > Server settings**. 
No need to modify dhcpcd.conf manually. However, manual edits will be reflected in the UI.

This topic is widely discussed in several past issues. See for example https://github.com/RaspAP/raspap-webgui/issues/335#issuecomment-541436840

**Comment by aluminum-ice** [comment, marked as answer]
> Metric values can be set for any available interface in **DHCP Server > Server settings**. 
> 
> No need to modify dhcpcd.conf manually. However, manual edits will be reflected in the UI.
> 
> 
> 
> This topic is widely discussed in several past issues. See for example https://github.com/RaspAP/raspap-webgui/issues/335#issuecomment-541436840

It didn't register for me until I dug in that the problem was the precedence of the routes. The page for managing the DHCP Server settings I find a bit confusing to be honest; not intuitive. But I do see that my manual edit is reflected in the interface. 

Great piece of software. I might become an Insider to help fund its further development. I'm using this as a travel router for sharing connections on planes and hotels.

**Reply by bhboyle** [reply]
Thank you. 

This discussion help me a great deal.

I have disabled the "Install a default route for this interface" on the DHCP server page and I am now able to use RaspAP and the setting survives a reboot.

I wonder if this setting should be added to the FAQ section about Wifi to Wfifi use.

Thank you again!

Cheers.

**Reply by billz** [reply]
@bhboyle glad to hear it. 

Re: FAQ, I agree and would take this a step further. I've been thinking of creating a page devoted to common use cases for RaspAP, this being one of them. eg., "for a wifi ~~extender~~ repeater, set these options" sort of thing. I'll carve out some time and work on this.

@shinigami-78 the DHCP page has grown in complexity, it's true. One of the challenges is balancing feature requests (eg., RaspAP should expose DHCP setting 'x') with usability. Compared with UIs such as OpenWRT and others I think we're far more comprehensible, perhaps at the risk of frustrating power users that want to administer every possible service option. At the same time we recognize that users of RaspAP, and general-purpose SBCs versus application-specific network routers generally, have different needs and expectations.

One idea I've put forward is tooltips and/or other forms of context sensitive help https://github.com/RaspAP/raspap-webgui/issues/872
My thinking is this combined with a "start here" type of page for the docs would go a long way toward addressing this. Thanks for your input :)

**Reply by turdferguson68** [reply]
> I have disabled the "Install a default route for this interface" on the DHCP server page and I am now able to use RaspAP and the setting survives a reboot.

Did you disable the default route for your client side, the hotspot side, or both>

**Reply by billz** [reply]
As promised 😉  https://docs.raspap.com/repeater/

**Comment by bhboyle** [comment]
In the DHCP server tab on the menu there is a button that is called "Install a default route for this interface" I turned that off. That is the only setting I changed and routed started working.

I would imagine that turning off the default route on the Hotspot side would stop it from routing data to the internet.

Hope this helps.

**Reply by billz** [reply]
@bhboyle I agree re: disabling this option. It's caused confusion for other users. In the meantime, I've updated the FAQ accordingly https://docs.raspap.com/faq/#interfaces 

Thanks for starting this thread.
